app/utils/hash_utils.rb

```diff
@@ -109,4 +109,28 @@ def transpose(x)
       acc
     }.map { |(k, v)| [k, v.to_a] }.to_h
   end
+
+  # { a: b: 1 } -> { :"a.b" => 1 }
+  def flatten(h)
+    # use helper lambda
+    acc = ->(prefix, hash) {
+      hash.inject({}) { |memo, (k, v)|
+        key_s = k.to_s
+
+        if !k.is_a?(Symbol) || key_s.include?(".")
```

Comment:
Text says key must be a String even though the test is that key must be a symbol.